core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1842,33 +1842,29 @@ public void delete(AlluxioURI path, DeleteContext context)
               LockPattern.WRITE_EDGE);
       try (LockedInodePath inodePath = mInodeTree
               .lockInodePath(lockingScheme)) {
-        mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+        mPermissionChecker.checkParentPermission(Mode.Bits.WRITE_EXECUTE, inodePath);
```

Comment:
Actually in many other places we are checking the parent `WRITE` perm instead of `WRITE_EXECUTE`. This does not quite conform to the POSIX standard https://stackoverflow.com/questions/54622606/what-permissions-are-needed-to-delete-a-file-in-unix

Reply:
https://github.com/Alluxio/alluxio/issues/15187